### DISCUSSION TITLE
Tune autocomplete?

### DISCUSSION THREAD

**Discussion by cmdcolin** [VonHeikemen/lsp-zero.nvim#31] [Q&A]
Hi there
I am a big fan of this plugin! I was wondering if there were any settings to tweak for autocomplete, or even if we should consider tweaking them by default

Example: I have some gobbledegook (a coded URL) in a markdown file

file.md:
```
<!--

eJztVctu2kAU_RU0q1Yixjbh5V0e5NFSioA2UqsKje2LPYk9tmYGSIL8770zJsYkRF1U6qoLFtzXOXN8Zu6WcJoC8cgYNo0ZSMky3nBt1z6xeyd2d-7aXsf17IFlO70fpElSKiLGiWc3SSjoBsQdC1VMvHb_tEnWDDaSeD-3hIU48u75Sn1yJ_c-9oVM5gl9GpdgceQMMKiecv1vxDhQcQ08S-E7jsBMtlxKUJNH4jnt027X6nR7_U7HcQY9u9Mkfj4BYZLVYAinECH3El7Acofk4DCpqFCGMnDk5Z4O3I7ddbFZwBqEBAwuaSKhSaiUkPrJAc_iFzIVNHioHe2b0x9GjjMc7U9xlrCIp8CVnOtiTMTAohiB3QFqE2R8yaKVoApZYjmP0kRg0YQlsMrLFq8SbhEl12rUH-7H1-v2ox3bfjN6e7zFHOE2fIFe5CaLiZ0BZp_Pp-1GTgOfZY0P4-svo-nHRlVUF0brUEqDygRUQZSJJx0zI3SQhjRXIGpczml6tgvi96PpKAsquivBtNZK5dJrtWTboil9zjjdSCvI0ta9L7KNBCsTUSsyJpEtjd4SQEO5wKPIB1-0rSVdmJOd2JZrtRcpzXMILcQiRZMwHsKjBkv-ITD-GCmKQhsN8QV6TcOW32W6C735YFUC-2br6AqoWgk4Un4kWRi02XhykaGzaQSHxkqSYJjOzmTK98Z6U7x316D_vrmOtL12mOR5sCv5g80OK__WazVqe8_Jlf_flQeufG3LmmzHzHYka2ZISCBQ-v3VUcYjLNZnk3G2eekgnhIrKGOlycuIflpjFsINUtdou1d4H_qK_XqrVCnjsZmBzMS8pBYzBBFBzAKakBLEWHJEfUhkHfsCn2cQet_sBmoC1R6LQBkh6vPmdbzqHr1fUV2rm_dKLmtwRN-vJKG52UFbVG3JEmQ4h0elZSxX6sEq1ZLTQLE1XL7iXZjLyhEEjXTLcevxAEyi-A3q17Nj
-->

Hello
```

I type "Hello" in the document, and it pops up an autocomplete suggesting that whole string be autocompleted for some reason

Then, kind of the second level of weirdness is that hitting "Enter" automatically chooses that autocompletion and it is filled in. To avoid this, I generally
1. hit escape to exit insert mode to make autocomplete go away
2. hit A to resume insert mode at the end of line
3. hit enter again to go to next line

Should it require a tab to choose the autocomplete setting? Is there a reason why it is trying to match this as autocompletion?

**Comment by cmdcolin** [comment]
for some reason github images isn't working but here is the basic config showing the autocompletion and there are no lsps loaded according to LspInfo https://imgur.com/a/AkgU4Zi

**Reply by VonHeikemen** [reply]
So, there was a bug there. It's fixed now. However, the default behaviour of selecting the first item will remain, but that can be configured. To prevent the automatic selection of the first item do this:

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

lsp.setup_nvim_cmp({
  completion = {completeopt = 'menu,menuone,noinsert,noselect'}
})

lsp.setup()
```

If there is no items selected you can press enter and it will insert a newline.

Do note that you can hide the completion menu using `ctrl + e`.

**Reply by cmdcolin** [reply]
awesome, thanks for the tips!

**Reply by cmdcolin** [reply]
I think it is continuing to try to autocomplete the large string but this workaround is helpful
